tapiro/core/src/main/scala/io/buildo/tapiro/AkkaHttpMeta.scala

```diff
@@ -40,17 +41,26 @@ object AkkaHttpMeta {
     q"pathPrefix($pathName) { List(..$rest).foldLeft[Route]($first)(_ ~ _) }"
   }
 
-  val endpoints = (routes: List[Route]) =>
-    routes.flatMap { route =>
-      val name = Term.Name(route.name.last)
-      val endpointsName = Term.Select(Term.Name("endpoints"), name)
-      val controllersName = Term.Select(Term.Name("controller"), name)
+  val endpoints = (routes: List[TapiroRoute]) =>
+    routes.map { route =>
+      val name = Term.Name(route.route.name.last)
+      val endpointsName = q"endpoints.$name"
+      val controllersName = q"controller.$name"
       val controllerContent =
-        if (route.params.length <= 1) Some(controllersName)
-        else Some(Term.Select(Term.Eta(controllersName), Term.Name("tupled")))
-      controllerContent.map { content =>
-        val toRoute = Term.Apply(Term.Select(endpointsName, Term.Name("toRoute")), List(content))
-        q"val ${Pat.Var(name)} = $toRoute"
-      }
+        route.method match {
+          case RouteMethod.GET =>
+            route.route.params.length match {
+              case 0 => q"_ => $controllersName()"
+              case 1 => controllersName
+              case _ => q"($controllersName _).tupled"
+            }
+          case RouteMethod.POST =>
+            val fields = route.route.params
+              .filterNot(_.tpe == MetarpheusType.Name("AuthToken"))
+              .map(p => Term.Name(p.name.getOrElse(Meta.typeNameString(p.tpe))))
+            q"x => $controllersName(..${fields.map(f => q"x.$f")})"
+        }
+      val toRoute = q"$endpointsName.toRoute($controllerContent)"
+      q"val ${Pat.Var(name)} = $toRoute"
```

Comment:
minor, but I think this can be

```suggestion
      q"val $name = $toRoute"
```

Reply:
I can't get this to work 🙁 it expects a `Pat` there and `$name` is a `Term.Name`, which doesn't seem to work (using a `String` instead compiles but generates a literal string pattern)